community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1923,26 +1939,62 @@ def nodeset_map(self, hostnames: list):
             nodeset_map[self.node_nodeset_name(node)].append(node)
         return nodeset_map
 
+    @lru_cache
+    def get_jobs(self) -> List[Job]:
+        res = run(f"{self.scontrol} show jobs", timeout=30)
+        all_jobs = [x.split() for x in res.stdout.split("\n\n")[:-1]]
+
+        res_jobs: List[Job] = []
+        for job in all_jobs:
+            for item in job:
+                key, value = item.split('=', 1) # Split at the first '='
+                if key == 'JobId':
+                    res_jobs.append(self.job(value))
```

Comment:
This wold cause multiple calls to `scontrol`, can't we extract information from returned result already?

Reply:
Ack, moved the common logic to separate internal function.